server/container/guice/mailet/src/main/java/org/apache/james/utils/GuiceGenericLoader.java

```diff
@@ -19,24 +19,77 @@
 
 package org.apache.james.utils;
 
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.james.filesystem.api.FileSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Injector;
 
 public class GuiceGenericLoader<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GuiceGenericLoader.class);
+    public static final String INCLUDED_JARS_FOLDER_NAME = "included-jars/";
```

Comment:
We can realy on streams toArray, which involves a cast (the size of the array being unknown)